pkg/common/oras/authprovider/azure/azureworkloadidentity.go

```diff
@@ -205,6 +222,56 @@ func (d *WIAuthProvider) Provide(ctx context.Context, artifact string) (provider
 	return authConfig, nil
 }
 
+// validateEndpoints checks if the endpoints are valid for auth provider.
+// A valid endpoint is either a fully qualified domain name or a wildcard domain
+// name folloiwing RFC 1034.
+// Valid examples:
+// - *.example.com
+// - example.com
+//
+// Invalid examples:
+// - *
```

Comment:
This is actually a valid case. `localhost` matches it, doesn't it?

Reply:
Hold-on. Let me double check

Reply:
according to the definition: `*.<anydomain>`, I feel it does not include * case. And do we need to support this in Azure scenario?

Reply:
Also, `*` itself is misleading and not meaningful as there is no single administrative entity for `*`. Let's keep it as invalid.

Reply:
https://serverfault.com/questions/1119905/are-wildcard-certificates-allowed-for-a-top-level-domain makes sense to me.